baseplate/experiments/__init__.py

```diff
@@ -134,25 +138,43 @@ def variant(self, name, user=None, bucketing_event_override=None,
         :rtype: :py:class:`str`
         :return: Variant name if a variant is active, None otherwise.
         """
+
         experiment = self._get_experiment(name)
 
         if experiment is None:
             return None
 
         inputs = dict(kwargs)
+
+        """ clean up user id. If not a baseplate user, we risk inconsistency."""
         if user:
             if user.is_logged_in:
                 inputs["user_id"] = user.id
                 inputs["user_roles"] = user.roles
+                inputs["logged_in"] = user.is_logged_in
             else:
                 inputs["user_id"] = user.loid
-            inputs["logged_in"] = user.is_logged_in
+                inputs["logged_in"] = False
+
+        #loid is numeric. Convert it to a base-36 string; Ideally, this will
+        #   be fixed before the caller sends it in, but this will catch
+        if isinstance(inputs.get("user_id"), int):
+            if (inputs.get("user_id") < 0):
```

Comment:
I don't see this kind of "warn and null out" pattern going on much in the rest of this code. Is that the sort of behaviour we want to use here?

nit: at this point we know that `inputs["user_id"]` is an int so the `.get()` guard on dictionary access is unnecessary.

style nit: no parens around the condition in python